packages/core/utils/lib/convert-query-params.js

```diff
@@ -218,7 +218,81 @@ const convertFieldsQueryParams = (fields, depth = 0) => {
   throw new Error('Invalid fields parameter. Expected a string or an array of strings');
 };
 
-const convertFiltersQueryParams = filters => filters;
+const convertFiltersQueryParams = (filters, schema) => {
+  // Filters need to be either an array or an object
+  // Here we're only checking for 'object' type since typeof [] => object and typeof {} => object
+  if (!isObject(filters)) {
+    throw new Error('The filters parameter must be an object or an array');
+  }
+
+  // Don't mutate the original object
+  const filtersCopy = cloneDeep(filters);
```

Comment:
I think this might cause an issue with Dates if we call it programmatically.

Reply:
What kind of issue do you have in mind?

From the tests I've made, it looks like dates are handled well handled.

(in this screenshot dates are stringified Date instances but in fact a new Date object is created with the value of the original one `cloneDeep(o).a.d[0].e === o.a.d[0].e` returns false) 
![image](https://user-images.githubusercontent.com/25851739/152139944-960ce43b-24a6-488e-bf99-4428b40bba5d.png)